proxy/auth_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package proxy
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+
+	"io/ioutil"
+
+	"github.com/openfaas/faas-cli/config"
+)
+
+func Test_SetAuth_AuthorizationHeader(t *testing.T) {
+	//setup store
+	config.DefaultDir, _ = ioutil.TempDir("", "faas-cli-auth-test")
+	config.DefaultFile = "authtest1.yml"
+	gatewayURL := strings.TrimRight("http://openfaas.test/", "/")
+	config.UpdateAuthConfig(gatewayURL, "Aladdin", "open sesame")
+
+	req, _ := http.NewRequest("GET", "http://openfaas.test/", nil)
+	SetAuth(req, gatewayURL)
+	header := req.Header.Get("Authorization")
+	expected := "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=="
+	if header != expected {
+		t.Errorf("got header %q, want %q", header, expected)
+	}
+}
+
+func Test_SetAuth_SkipAuthorization(t *testing.T) {
+	//setup store
+	config.DefaultDir, _ = ioutil.TempDir("", "faas-cli-auth-test")
+	config.DefaultFile = "authtest2.yml"
+	gatewayURL := strings.TrimRight("http://openfaas.test/", "/")
+	config.UpdateAuthConfig(gatewayURL, "Aladdin", "open sesame")
```

Comment:
I like this test a lot but I had to read it 3-4 times. I think it could be easier to read if we broke out the URLs into variables so it's easier to read and understand at a glance.

```
basicAuthURL
openURL
```